### PR TITLE
NAS-109283 / 21.02 / Better handle system managed internal datasets

### PR DESCRIPTION
**Pull request by sonicaj**
This PR fixes following issues:

1) We make sure to remove system managed internal datasets from children attribute of a dataset.
2) We make sure that we are not able to create a dataset under system managed internal datasets using `pool.dataset.create` api.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -2765,28 +2765,30 @@ def query(self, filters, options):
...
-        filters.extend([
+        children_filters.extend([
             ['id', 'rnin', '.glusterfs'],
         ])
 
         if osc.IS_LINUX:
             k8s_config = self.middleware.call_sync('kubernetes.config')
             if k8s_config['dataset']:
-                filters.append(['id', '!^', f'{k8s_config["dataset"]}/'])
+                children_filters.append(['id', '!^', f'{k8s_config["dataset"]}/'])
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py:2789]: Do we exclude `k8s_config["dataset"]` itself?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2789]: Nope, we still show it

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2789]: The attachment delegate should handle that case if the user intends to delete the dataset

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py]: Good point :)

**Review by themylogin — APPROVED** [review]
Great, thank you!

**Review by william-gr — APPROVED** [review]

**Pull request merged**